### PR TITLE
REST API payment events can now be properly filtered by token_address

### PR DESCRIPTION
**Pull request by LefterisJP**
## Description

Fixes: #5591 

Please, describe what this PR does in detail:
- If it's a bug fix, detail the root cause of the bug and how this PR fixes it.
- If it's a new feature, describe the feature this PR is introducing and design decisions.
- If it's a refactoring, describe why it is necessary. What are its pros and cons in respect to the previous code, and other possible design choices.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/api/python.py [hunk 1]
```diff
@@ -89,7 +91,12 @@
...
 
 
-def event_filter_for_payments(event: Event, partner_address: Address = None) -> bool:
+def event_filter_for_payments(
+    event: Event,
+    chain_state: Optional[ChainState] = None,
+    partner_address: Optional[Address] = None,
+    token_address: Optional[TokenAddress] = None,
+) -> bool:
     """Filters payment history related events depending on partner_address argument
```
Comment (palango) [inline raiden/api/python.py]: Can you update the docstring as well

File: raiden/api/python.py [hunk 2]
```diff
@@ -1124,10 +1143,16 @@ def get_raiden_events_payment_history_with_timestamps(
             logical_and=False,
         )
 
+        chain_state = chain_state = views.state_from_raiden(self.raiden)
```
Comment (palango) [inline raiden/api/python.py]: One assignment should be enough

Comment (LefterisJP) [reply raiden/api/python.py]: LOL WTF how come no linter complained about this

File: raiden/api/python.py (hunk 1 above)
Comment (LefterisJP) [reply raiden/api/python.py]: sure. Commit incoming.

**Pull request merged**